src/appshell/view/preferences/preferencesmodel.cpp

```diff
@@ -143,7 +143,11 @@ void PreferencesModel::load(const QString& currentPageId)
     if (!currentPageId.isEmpty()) {
         setCurrentPageId(currentPageId);
     } else {
-        setCurrentPageId("general");
+        if (configuration()->preferencesDialogLastOpenedPageId().isEmpty()) {
+            setCurrentPageId("general");
+        } else {
+            setCurrentPageId(configuration()->preferencesDialogLastOpenedPageId());
```

Comment:
Let's prevent calling `configuration()->preferencesDialogLastOpenedPageId()` twice:
```suggestion
        const QString& lastOpenedPageId = configuration()->preferencesDialogLastOpenedPageId();
        if (lastOpenedPageId.isEmpty()) {
            setCurrentPageId("general");
        } else {
            setCurrentPageId(lastOpenedPageId);
```

Reply:
Done